requirements.txt

```diff
@@ -2,3 +2,4 @@ requests~=2.32.3
 pysolr~=3.10.0
 setuptools~=47.1.0
 PySocks~=1.7.1
+smart_open[all]~=7.0.1
```

Comment:
While we just use `s3`, I chose to use `all` to account for the other providers. Let me know if that feels OK but it can be handled differently that's introducing to many potential dependencies